src/main/scala/rocket/RocketCore.scala

```diff
@@ -119,6 +124,7 @@ class Rocket(tile: RocketTile)(implicit p: Parameters) extends CoreModule()(p)
   val gated_clock =
     if (!rocketParams.clockGate) clock
     else ClockGate(clock, clock_en, "rocket_clock_gate")
+  val ALUClass = if (usingABLU) ABLU else ALU
```

Comment:
use `AbstractAlu` here.

Reply:
Does `useABLU` need to be its own parameter? Should `useABLU = useBitManip || useBitManipCrypto || useCryptoNIST || useCryptoSM`

Reply:
> Should `useABLU = useBitManip || useBitManipCrypto || useCryptoNIST || useCryptoSM`

Yes it is but conditioning on this everywhere is not maintainable.

I think the `useABLU` in `CoreParams` is not so necessary (I thought there might be some one just trying to use the ABLU alone, but yes generally it is not the case) but `usingABLU` in `HasCoreParameters` is useful for internal usage

Reply:
You don't need to condition on this everywhere. We shouldn't expose `useABLU` as a parameter in `RocketCoreParams` if its actually a derived field.
You can set `val useABLU = ...` in the body of `RocketCoreParams`

Reply:
> You can set `val useABLU = ...` in the body of `RocketCoreParams`

Fixed